arangodb/src/main/java/com/yahoo/ycsb/db/ArangoDBClient.java

```diff
@@ -0,0 +1,488 @@
+      arangoDriver.deleteDocument(table, key);
+      return Status.OK;
+    } catch (ArangoException e) {
+      if (e.getErrorNumber() != ErrorNums.ERROR_ARANGO_DOCUMENT_NOT_FOUND) {
+        logger.error("Fail to delete: " + table + " " + key,  e);
+      } else {
+        logger.debug("Trying to delete document not exist: {} {}", table, key);
+        return Status.NOT_FOUND;
+      }
+    } catch (Exception e) {
+      logger.error("Exception while trying delete " + table + " " + key, e);
+    }
+    return Status.ERROR;
+  }
+
+  /**
+   * Perform a range scan for a set of records in the database. Each
+   * field/value pair from the result will be stored in a HashMap.
+   * 
+   * @param table
+   *      The name of the table
+   * @param startkey
+   *      The record key of the first record to read.
+   * @param recordcount
+   *      The number of records to read
+   * @param fields
+   *      The list of fields to read, or null for all of them
+   * @param result
+   *      A Vector of HashMaps, where each HashMap is a set field/value
+   *      pairs for one record
+   * @return Zero on success, a non-zero error code on error. See the
+   *     {@link DB} class's description for a discussion of error codes.
+   */
+  @Override
+  public Status scan(String table, String startkey, int recordcount, Set<String> fields,
+      Vector<HashMap<String, ByteIterator>> result) {
+    DocumentCursor<BaseDocument> cursor = null;
+    try {
+      String aqlQuery = String.format(
+          "FOR target IN %s FILTER target._key >= @key SORT target._key ASC LIMIT %d RETURN %s ", table,
+          recordcount, constructReturnForAQL(fields, "target"));
+
+      Map<String, Object> bindVars = new MapBuilder().put("key", startkey).get();
+      cursor = arangoDriver.executeDocumentQuery(aqlQuery, bindVars, null, BaseDocument.class);
+      Iterator<BaseDocument> iterator = cursor.entityIterator();
+      while (iterator.hasNext()) {
+        BaseDocument aDocument = iterator.next();
+        HashMap<String, ByteIterator> aMap = new HashMap<String, ByteIterator>(aDocument.getProperties().size());
+        this.fillMap(aMap, aDocument.getProperties());
+        result.add(aMap);
+      }
+      return Status.OK;
+    } catch (Exception e) {
+      logger.error("Exception while trying scan " + table + " " + startkey + " " + recordcount, e);
+    } finally {
+      if (cursor != null) {
+        try {
+          cursor.close();
+        } catch (ArangoException e) {
+          logger.error("Fail to close cursor", e);
+        }
+      }
+    }
+    return Status.ERROR;
+  }
+
+  private String createDocumentHandle(String collectionName, String documentKey) throws ArangoException {
+    validateCollectionName(collectionName);
+    return collectionName + "/" + documentKey;
+  }
+
+  private void validateCollectionName(String name) throws ArangoException {
+    if (name.indexOf('/') != -1) {
+      throw new ArangoException("does not allow '/' in name.");
```

Comment:
nit (feel free to fix or disregard): This might give clearer error messages if you used IllegalArgumentException here rather than ArangoException.